### DISCUSSION TITLE
Add tag support

### DISCUSSION THREAD

**Discussion by bourvill** [wallabag/ios-app#376] [General]
Discuss related to #301

**Comment by bourvill** [comment]
First at all button a context menu can't hold "long task/workflow" -  Context menu is "killed" after tap 

you need to add `@State var ...` in EntriesListView and bind in your tag button.

And you can show a sheet from this state with `TagListFor`
You can see an example of sheet + tag in `EntryView`

**Reply by satishsurath** [reply]
Got it. Thanks for hints - its very insightful. Let me digest your message and the source code.
Will post my progress here and come back with any questions.

**Comment by gabeklavans** [comment]
I'm thinking of picking this back up (assuming no progress has been made since the last post here)

My wishlist for tags includes:
- [ ] Edit tags from context menu in articles list (to avoid needing to open the article first to edit the tags)
- [ ] Filter by tag(s) in articles list
- [ ] Ability to add tags during share sheet import flow (similar to Omnivore's current flow)

There's also a bit of wonkiness when adding preexisting tags to an article that doesn't have tags already, but I may make a separate bug issue for that.

**Reply by grysom** [reply]
Hi,

I would love having the ability to add tags when saving a link using Share shortcut on iOS. 

Is that what you mean by "Ability to add tags during share sheet import flow"? 

Has there been any progress towards adding this ability?

**Reply by gabeklavans** [reply]
> Is that what you mean by "Ability to add tags during share sheet import flow"? 

Yup that's exactly what I mean. 

> Has there been any progress towards adding this ability? 

Nope, I've actually since switched to Readeck since their features meet my needs a bit better, at least for now. But adding those features and making the progress sync way more aggressive would make me consider Wallabag again, I just found an alternative before I fully made time to work on this 😁

**Reply by grysom** [reply]
Ah, I see. Thank you for responding! Readneck seems interesting. Am I right assuming that, at the moment, Readneck does not support saving bookmarks via share sheet on iOS?

**Reply by gabeklavans** [reply]
> Ah, I see. Thank you for responding! Readneck seems interesting. Am I right assuming that, at the moment, Readneck does not support saving bookmarks via share sheet on iOS?

The developer provides a shortcut that gets the job done:

https://readeck.org/en/docs/faq#is-there-an-android-or-ios-application

**Comment by utrenkner** [comment]
I had been a years long user of Pocket and switched to Wallabag at the beginning of this year.

What I really miss from the Pocket-app is the option to add tags at the time of sharing a URL to the Wallabag app. The pocket app saved the article and then shortly gave you the option to add tags (maybe 3 seconds).

If I want to add tags to a saved article I just shared to the Wallabag app, I have to switch to the app, update my saved articles, open the new article before I could add tags to it.

Because I often save articles I don't have the time to read now, I often don't have the time for this workflow either. I now have hundreds of saved articles without suitable tags.

I would be happy to contribute financially to such a feature in this app.

**Comment by bourvill** [comment]
It's not really a question of money, and I appreciate you wanting to have it this way.
But it's more a question of time.
I'll make it a priority for my next session on the Wallabag app.

**Reply by utrenkner** [reply]
Thank you, I appreciate it. I assumed that money was not the issue but wanted offer something.

**Comment by bourvill** [comment]
https://github.com/user-attachments/assets/2a7332da-7b07-4102-9bd0-39201680429b

I'm working on it!
